upgrade-2.x.sh

```diff
@@ -1231,12 +1253,22 @@ _prepare_locking
 # Try to get lock, and exit if cannot, meaning another instance is running already
 exlock_now || exit 9
 
-if [ -z "$target_version" ]; then
-    log ERROR "--hostos-version is required."
+# Enforce required arguments. See comment above at 'Script start'.
+if [ -z "$target_version" ] && [ -z "$app_uuid" ]; then
+    log ERROR "Either --hostos-version or --app-uuid is required."
 fi
-
-if [ -z "${REGISTRY_ENDPOINT}" ]; then
-    log ERROR "--balenaos-registry is required."
+if [ -n "$target_version" ] && [ -n "$app_uuid" ]; then
+    log ERROR "Only one of --hostos-version and --app-uuid may be specified."
+fi
+if [ -n "$target_version" ]; then
+    if [ -z "${REGISTRY_ENDPOINT}" ]; then
```

Comment:
I believe these nested ifs can be combined no ?


```suggestion
if [ -n "$target_version" ] && [ -z "${REGISTRY_ENDPOINT}" ]; then
```

Reply:
This sequence of conditionals is based a choice of 1) target version + registry host or 2) app UUID + release commit. I think the logic is a little clearer if they are separated, but I'm happy to combine them as well.

Reply:
I don't have a strong opinion on it, if you find it more readable it's OK with me.